### PR TITLE
[android] - use FocalPoint for zoom

### LINKED ISSUE #8413: ZoomButtonController not zooming to center
ezgif com-video-to-gif 18

### PR DESCRIPTION
**Pull request by tobrun**
This PR fixes #8413 by taking in account focal point configuration or use the center of the screen when performing zoom transformations. This removes the `-1` default value from the codebase and replaces it with a concrete value as soon as possible (inside the interfaces that get called as a result of an action). This simplifies the code in `Transfrom.java` and `NativeMapView.java`.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java [hunk 1]
```diff
@@ -939,12 +942,57 @@ public void onRegisterFlingListener(MapboxMap.OnFlingListener listener) {
...
+
+    MapZoomControllerListener(MapGestureDetector detector, UiSettings uiSettings, Transform transform) {
+      this.mapGestureDetector = detector;
+      this.uiSettings = uiSettings;
+      this.transform = transform;
+    }
+
+    // Not used
+    @Override
+    public void onVisibilityChanged(boolean visible) {
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java]: If it isn't used, should we apply the [interface-segregation principle (ISP)](https://en.wikipedia.org/wiki/Interface_segregation_principle)?
This is only a comment (no blocker at all), but something to take into account and fix in the future.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 2]
```diff
@@ -174,27 +174,23 @@ private long getDurationNano(long durationMs) {
     return cameraPosition.zoom;
   }
 
-  void zoom(boolean zoomIn) {
-    zoom(zoomIn, -1.0f, -1.0f);
-  }
-
-  void zoom(boolean zoomIn, float x, float y) {
+  void zoom(boolean zoomIn, float focalX, float focalY) {
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: We should start considering refactoring [data clumps](https://martinfowler.com/bliki/DataClump.html) like `focalX`/ `focalY`. Here we could [introduce a parameter object](https://refactoring.com/catalog/introduceParameterObject.html).

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java (hunk 1 above)
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java]: :+1:

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java (hunk 2 above)
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: :+1:

**Pull request merged**